include/util/coordinate_calculation.hpp

```diff
@@ -25,16 +25,49 @@ const constexpr long double EARTH_RADIUS = 6372797.560856;
 
 
 //! Takes the squared euclidean distance of the input coordinates. Does not return meters!
-std::uint64_t squaredEuclideanDistance(const Coordinate &lhs, const Coordinate &rhs);
+std::uint64_t squaredEuclideanDistance(const Coordinate lhs, const Coordinate rhs);
 
 double haversineDistance(const Coordinate first_coordinate, const Coordinate second_coordinate);
 
 double greatCircleDistance(const Coordinate first_coordinate, const Coordinate second_coordinate);
 
-std::pair<double, FloatCoordinate>
-projectPointOnSegment(const FloatCoordinate &projected_xy_source,
-                      const FloatCoordinate &projected_xy_target,
-                      const FloatCoordinate &projected_xy_coordinate);
+inline std::pair<double, FloatCoordinate> projectPointOnSegment(const FloatCoordinate &source,
+                                                         const FloatCoordinate &target,
+                                                         const FloatCoordinate &coordinate)
```

Comment:
Using `floats` yields numerical instabilities when doing conversion from/to fixed point representation. There was a bug report on this a while back, especially yielding to problems with high resolution map matched traces (all the coordinates are projected using this function).

We should benchmark if there is actually a difference between passing by value or const ref.